src/Microsoft.Data/GroupBy.cs

```diff
@@ -0,0 +1,432 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A GroupBy class that is typically the product of a DataFrame.GroupBy call
```

Comment:
(minor) instead of saying `the product of a DataFrame.GroupBy call`, I would go with `the result of a DataFrame.GroupBy call`.

Also, add a period to the end of the sentence.